language/go/generate.go

```diff
@@ -404,9 +404,8 @@ func emptyPackage(c *config.Config, dir, rel string, f *rule.File) *goPackage {
 		dir:  dir,
 		rel:  rel,
 	}
-	if err := pkg.inferImportPath(c); err != nil {
-		log.Printf("could not infer import path: %v", err)
-	}
+	// It's fine if it fails to infer import path
+	_ = pkg.inferImportPath(c)
```

Comment:
Is it the right thing to do to silently ignore this error? Why is it fine to ignore?

Reply:
See the PR summary. Let me know if it's still not clear or you have a better way to fix that.

Reply:
Exactly. The information necessary to understand the silencing of the error isn't in the code. A later reader is very unlikely to be able to figure out why this error shouldn't be printed based upon the context that's available here and might bring it back. Since there isn't a test case, we have to depend on inline documentation to describe our intent.

IMO comments should explain *why* things are happening, not what is happening -- the code describes more accurately what is happening.

Reply:
How about removing this function call all together?

Reply:
I think that's fine. `inferImportPath` does have a side effect

https://github.com/bazelbuild/bazel-gazelle/blob/1ab6bae363de3da4760501cb541fd80cb93031ec/language/go/package.go#L157-L167

Do we care about that though? I believe that it is just caching the import path inference, but want to double check that you think that's reasonable.

Assuming you think omission is safe, LGTM and let's merge it 👍

Reply:
The reason why this omission is safe is that later in the code, Gazelle checks whether `pkg.ImportPath` is set, and call `pkg.inferImportPath` when it's not: https://github.com/bazelbuild/bazel-gazelle/blob/32dab97bac88b1a35653d33a034b38e55d757983/language/go/generate.go#L164-L165

So the call I remove in this function is a duplicate call